common/lib/src/realm_common_base.dart

```diff
@@ -16,12 +16,36 @@
 //
 // //////////////////////////////////////////////////////////////////////////////
 
+/// An enum controlling the base type for a [RealmModel].
+///
+/// {@category Annotations}
+enum ObjectType {
+  /// A standalone top-level object that can be persisted in Realm. It can link
+  /// to other objects or collections of other objects.
+  topLevel('RealmObject'),
+
+  /// An object that can be embedded in other objects. It is considered owned
+  /// by its parent and will be deleted if its parent is deleted.
+  embedded('EmbeddedObject'),
+
+  /// A special type of object used to facilitate unidirectional synchronization
+  /// with Atlas App Services. It is used to push data to Realm without the ability
+  /// to query or modify it.
+  _asymmetric('AsymmetricObject');
+
+  const ObjectType([this.className = 'Unknown']);
+
+  /// The name of the base class exposed by the Realm SDK.
+  final String className;
```

Comment:
It works well when converting to code - i.e. https://github.com/realm/realm-dart/blob/af2aafca1ab5b3612ea2414ffca016fa300a7649/generator/lib/src/realm_model_info.dart#L35, but of course we could also use an int, I don't have strong opinions. I imagine by "you don't need an extension" you're referring to https://github.com/realm/realm-dart/blob/af2aafca1ab5b3612ea2414ffca016fa300a7649/lib/src/native/realm_core.dart#L2630? In this case, I wanted to keep it as an extension because I wanted to make sure we're using the generated constants from `realm_class_flags` rather than hardcoding some values and making sure they're always in sync.